src/drawing/DrawingCreator.tsx

```diff
@@ -0,0 +1,163 @@
+    onStop,
+    stroke = defaultStroke,
+}: Props): JSX.Element {
+    const [drawingStatus, setDrawingStatus] = React.useState<DrawingStatus>(DrawingStatus.init);
+    const capturedPathRef = React.useRef<Array<Position>>([]);
+    const creatorElRef = React.useRef<HTMLDivElement>(null);
+    const drawingDirtyRef = React.useRef<boolean>(false);
+    const drawingPathRef = React.useRef<DrawingPathRef>(null);
+    const drawingSVGRef = React.useRef<DrawingSVGRef>(null);
+    const renderHandleRef = React.useRef<number | null>(null);
+
+    const getPathGroup = (): PathGroup | null => {
+        const { current: creatorEl } = creatorElRef;
+        const { current: points } = capturedPathRef;
+
+        if (!creatorEl || points.length <= 1) {
+            return null;
+        }
+
+        return {
+            clientId: uuid.v4(),
+            paths: [{ clientId: uuid.v4(), points }],
+            stroke,
+        };
+    };
+    const getPosition = (x: number, y: number): [number, number] => {
+        const { current: creatorEl } = creatorElRef;
+
+        if (!creatorEl) {
+            return [x, y];
+        }
+
+        // Calculate the new position based on the mouse position less the page offset
+        // and convert to a percentage of the page
+        const { height, left, top, width } = creatorEl.getBoundingClientRect();
+        return [((x - left) / width) * 100, ((y - top) / height) * 100];
+    };
+
+    // Drawing Lifecycle Callbacks
+    const startDraw = (x: number, y: number): void => {
+        const [x1, y1] = getPosition(x, y);
+
+        setDrawingStatus(DrawingStatus.dragging);
+
+        capturedPathRef.current = [{ x: x1, y: y1 }];
+        drawingDirtyRef.current = true;
+    };
+    const stopDraw = (): void => {
+        const pathGroup = getPathGroup();
+
+        setDrawingStatus(DrawingStatus.init);
+
+        capturedPathRef.current = [];
+        drawingDirtyRef.current = true;
+
+        if (pathGroup) {
+            onStop(pathGroup);
+        } else {
+            onAbort();
+        }
+    };
+    const updateDraw = (x: number, y: number): void => {
+        const [x2, y2] = getPosition(x, y);
+        const { current: points } = capturedPathRef;
+
+        points.push({ x: x2, y: y2 });
+        drawingDirtyRef.current = true;
+
+        if (drawingStatus !== DrawingStatus.drawing) {
+            setDrawingStatus(DrawingStatus.drawing);
+            onStart();
+        }
+    };
+
+    // Event Handlers
+    const renderStep = (callback: () => void): void => {
+        renderHandleRef.current = window.requestAnimationFrame(callback);
+    };
+    const renderPath = (): void => {
+        const { current: creatorEl } = creatorElRef;
+        const { current: points } = capturedPathRef;
+        const { current: isDirty } = drawingDirtyRef;
+        const { current: svgPath } = drawingPathRef;
+
+        if (!creatorEl) {
+            return;
+        }
+
+        const d = getPathCommands(points);
```

Comment:
We should avoid try to avoid variable name abbreviations, even if the attribute itself is abbreviated.